classes/migration/upgrade/v3_4_0/I7191_InstallSubmissionHelpDefaults.php

```diff
@@ -0,0 +1,137 @@
+        if (empty($this->CONTEXT_TABLE) || empty($this->CONTEXT_SETTINGS_TABLE) || empty($this->CONTEXT_COLUMN)) {
+            throw new Exception('Upgrade could not be completed because required properties for the I7191_InstallSubmissionHelpDefaults migration are undefined.');
+        }
+
+        $initialLocale = Locale::getLocale();
+
+        DB::table($this->CONTEXT_TABLE . ' as ct')
+            ->leftJoin($this->CONTEXT_SETTINGS_TABLE . ' as cst', function (JoinClause $join) {
+                $join->on('ct.' . $this->CONTEXT_COLUMN, '=', 'cst.' . $this->CONTEXT_COLUMN)
+                    ->where('cst.setting_name', '=', 'supportedLocales');
+            })
+            ->get([
+                'ct.' . $this->CONTEXT_COLUMN,
+                'ct.path',
+                'ct.primary_locale',
+                'cst.setting_value as supportedLocales'])
+            ->each(function (stdClass $row) {
+                foreach (json_decode($row->supportedLocales) as $locale) {
+                    Locale::setLocale($locale);
+                    $localizationParams = $this->getLocalizationParams($row, $locale);
+                    DB::table($this->CONTEXT_SETTINGS_TABLE)
+                        ->insert(
+                            $this->getNewSettings()->map(
+                                function ($localeKey, $settingName) use ($row, $locale, $localizationParams) {
+                                    return [
+                                        $this->CONTEXT_COLUMN => $row->{$this->CONTEXT_COLUMN},
+                                        'locale' => $locale,
+                                        'setting_name' => $settingName,
+                                        'setting_value' => __($localeKey, $localizationParams),
+                                    ];
+                                }
+                            )->toArray()
+                        );
+                }
+            });
+
+
+        Locale::setLocale($initialLocale);
+    }
+
+    public function down(): void
+    {
+        DB::table($this->CONTEXT_SETTINGS_TABLE)
+            ->whereIn('setting_name', $this->getNewSettings()->keys())
+            ->delete();
+    }
+
+    /**
+     * @return Collection [settingName => localeKey]
+     */
+    protected function getNewSettings(): Collection
+    {
+        return collect([
+            'beginSubmissionHelp' => 'default.submission.step.beforeYouBegin',
+            'contributorsHelp' => 'default.submission.step.contributors',
+            'detailsHelp' => 'default.submission.step.details',
+            'forTheEditorsHelp' => 'default.submission.step.forTheEditors',
+            'reviewHelp' => 'default.submission.step.review',
+            'uploadFilesHelp' => 'default.submission.step.uploadFiles',
+        ]);
+    }
+
+    /**
+     * Get the params needed to localize some of the default settings
+     */
+    protected function getLocalizationParams(stdClass $row, string $locale): array
```

Comment:
If this wasn't migration code, and if it didn't introduce just as many complications as it would solve, I'd suggest unpacking the `stdClass` into named parameters to allow the function to better self-document and enforce parameter requirements. But this is just migration code, and it would require the `->get` list to alias columns to match the function requirements, which is pretty wonky, so never mind.